runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -532,7 +532,7 @@ static bool scanForNativeMethodsUntilMonitorNode(TR::TreeTop *firstTree, TR::Com
          {
          TR::ResolvedMethodSymbol *symbol = callTestNode->getSymbol()->castToResolvedMethodSymbol();
          if (strstr(symbol->signature(comp->trMemory()), "java/lang/Object.notify") ||
-             strstr(symbol->signature(comp->trMemory()), "java/lang/Object.wait"))
+             strstr(symbol->signature(comp->trMemory()), "java/lang/Object.waitImpl"))
```

Comment:
I wonder if the check needs to be for `Object.wait` OR `Object.waitImpl` to be consistent with the original semantics when `Object.wait` was native.  If `Object.wait` is not inlined by the JIT, the semantics are the same as before.  But if `Object.wait` is inlined then we want the check to be for the native `Object.waitImpl` method.

Reply:
based on the code that sets `callTestNode`, it is checking for resolved native node, so the original code is definitely looking for the native method, I don't know if there is special handling on the JIT side where `Object.wait` is recognized and inlined differently than normal methods (I wasn't able to find such code in OpenJ9 though)

Reply:
I'm pasting this here because I'm tracing the history of this code to get the context.  It was a performance feature introduced early days in lock reservation implementation, motivated thusly:
```
This is a problem with excessive cancellations happening while
doing notify and wait on an object that was reserved. We shouldn't 
reserve objects that we do notify or wait on
```
We'll have to make sure that the semantics are preserved.  Still looking into it...

Reply:
I think the `strstr` would match the new native without changing the code here. It's not matching signatures here, so it's trying to catch all of the wait methods in one check.

Also, `waitImpl` is private, so it will never be called directly, only from the other `wait` implemenatations.

I suggest removing this part of the change entirely.

Reply:
In fact, this change is clearly incorrect given the matching using `strstr`. It absolutely has to be removed.

Reply:
fixed